### DISCUSSION TITLE
It's really the best RTC encoder. Amazing!

### DISCUSSION THREAD

**Discussion by shennungshi** [pigpeppa/xin26x#5] [General]
So fast and have a good performance. 
Cool!

**Comment by urnotlcxz** [comment]
I want to know how can you run it？I open it in VS Code also use Encoder Build or not that open the _xin26x_test.exe_ directly.  I also can't find the place to load the input video file.

**Comment by pigpeppa** [comment]
Please refer to Building Project in ReadMe.

If you would like to build xin26x project, you have to install CMAKE and VS. I am using VS2013.

As for loading input video file, command line -i is for loading input YUV. We accept YUV420 8 bit sequence.

If you do not have YUV sequence, you can download YUV file from  https://media.xiph.org/video/derf/

**Reply by urnotlcxz** [reply]
Thanks！I build _make-solutions.bat_ successfully, and what should I do next step? For example, to run the _xin26x_test.exe_? or run which .c file or some else files?
 I also have downloaded the YUV file according to your website: click the first "akiyo"  QCIF file _akiyo_qcif.y4m_ 
Can I directly use this to replace _input.yuv_?  And should I still use the default parameters which in _fast.cfg_ file or make some changes? If not, I saw that website shows that in **source link** the QCIF Format(176x144),CIF Format(352x288), and in **readme** file it shows Y_RES 720X486, UV_RES 360X486. Which is the right size can config?
Thanks for your replying!

**Reply by pigpeppa** [reply]
Currently, Xin26x accept YUV 420 sequence, so you can convert *.y4m to *.yuv by ffmpeg.

**Reply by urnotlcxz** [reply]
Thanks, I think maybe I‘m on the way to success. I build it and use `xin26x_test.exe -i input.yuv -w 176 -h 114 -f 30 -a 2 -p 1 -r 0 -q 32 -o test.bin`
But it only shows the 

>  xinConfig:
InputFileName =
OutputFileName =
ReconFileName =
inputWidth = 176
inputHeight = 144
STATUS: Xin26x lib build date: Feb 23 2021 14:36:49.
 Start Coding......

and the test.bin is 0KB. What steps or procedures did I miss?

**Comment by huhu727** [comment]
I am wondering if xin26x supports yuv420 10bit depth file....

**Comment by pigpeppa** [comment]
no, it currently does not support 10bit.



***@***.***
 
From: Lazy-Koala-cuc
Date: 2023-01-18 19:45
To: pigpeppa/xin26x
CC: Pig Peppa; Comment
Subject: Re: [pigpeppa/xin26x] It's really the best RTC encoder. Amazing! (Discussion #5)
I am wondering if xin26x supports yuv420 10bit depth file....

**Reply by huhu727** [reply]
Thank you for your replying. No wonder I get error message while I input 10bit yuv file. 😂
